x-pack/plugins/observability_solution/observability_onboarding/public/application/shared/logo_icon.tsx

```diff
@@ -66,10 +68,47 @@ function useIconForLogo(logo?: SupportedLogo): string | undefined {
   }
 }
 
-export function LogoIcon({ logo, size }: { logo: SupportedLogo; size?: EuiIconProps['size'] }) {
+function toAvatarSize(size?: EuiIconProps['size']): EuiAvatarProps['size'] {
+  switch (size) {
+    case 's':
+      return 's';
+    case 'm':
+      return 'm';
+    case 'l':
+      return 'l';
+    case 'xl':
+      return 'xl';
+    default:
+      return 'xl';
+  }
+}
```

Comment:
Essentially just making TypeScript happy because you cannot provide an `EuiIconProps` to `EuiAvatarProps` size as they have different values defined.

Basically, we want to accept `xxl` as a size for icons, but `EuiAvatar` doesn't support this size, so in cases where we want the avatar we can only accept `xl`. Maybe better to avoid supplying the EUI types, or inverting the rendering by accepting an avatar from the parent.

Reply:
Wouldn't it be easier to just restrict the type of your `LogoIcon` components' `size` prop accordingly instead of trying to convert between the two? 

If it's important that the prop works for both components (`EuiIcon` and `EuiAvatar`) you can use the intersection: `EuiIconProps['size'] & & EuiAvatarProps['size']`. 

If the prop is only used by `EuiAvatar` then simply use `EuiAvatarProps['size']`

Reply:
Well, the parent component accepts a `size` prop that was previously only `EuiIconProps['size']`, but now we need to provide that to the Avatar component, whose size is a subset. I think you're right we should accept an intersection and we can use a [type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) to infer the right type in the parent.

Reply:
Let me know your thoughts on https://github.com/elastic/kibana/pull/188340/commits/db27ddecfc7ee9b9be3391a492204d08f0e19d45 and if you think we can further clean it up.

Reply:
Looks great, much simpler!